llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -334,6 +334,13 @@ Value *VPTransformState::get(VPValue *Def, unsigned Part) {
   return VectorValue;
 }
 
+Value *VPTransformState::get(VPValue *Def, unsigned Part,
+                             unsigned NeedsScalar) {
```

Comment:
There's a slight redundancy when a vector is needed - either pass `false` or call get(Def, Part).
Perhaps worth extending get(Def, Part) to accept an optional `bool NeedsScalar = false`, and have it redirect to return get(Def, VPIteration(Part, 0))?